File: apps/mobile/modules/launch-intent/android/src/main/java/expo/modules/launchintent/LaunchIntentStore.kt
package expo.modules.launchintent

object LaunchIntentStore {
  private var pendingUrl: String? = null
  private var module: LaunchIntentModule? = null

  fun attach(module: LaunchIntentModule) {
    this.module = module
  }

  fun detach(module: LaunchIntentModule) {
    if (this.module === module) {
      this.module = null
    }
  }

  fun setPendingUrl(url: String?) {
    val normalized = url?.trim()?.takeIf { it.isNotEmpty() } ?: return
    pendingUrl = normalized
    module?.emitUrl(normalized)
  }

  fun peekPendingUrl(): String? {
    return pendingUrl
  }

  fun consumePendingUrl(): String? {
    val value = pendingUrl
    pendingUrl = null
    return value
  }
}

File: apps/mobile/modules/launch-intent/ios/LaunchIntentStore.swift
import Foundation

final class LaunchIntentStore {
  static let shared = LaunchIntentStore()

  private weak var module: LaunchIntentModule?
  private var pendingUrl: String?

  private init() {}

  func attach(module: LaunchIntentModule) {
    self.module = module
  }

  func detach(module: LaunchIntentModule) {
    if self.module === module {
      self.module = nil
    }
  }

  func setPendingUrl(_ url: String?) {
    guard let normalized = url?.trimmingCharacters(in: .whitespacesAndNewlines), !normalized.isEmpty else {
      return
    }

    pendingUrl = normalized
    module?.emitUrl(normalized)
  }

  func peekPendingUrl() -> String? {
    return pendingUrl
  }

  func consumePendingUrl() -> String? {
    let value = pendingUrl
    pendingUrl = nil
    return value
  }
}

